<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="decim_fac" value="4" />

    <!-- Image processing nodelets -->
    <node pkg="nodelet" type="nodelet" name="image_proc_nodelet_manager" args="manager" />

    <!--Downsampling -->
    <node pkg="nodelet" type="nodelet" name="image_proc_downsample" args="load image_proc/crop_decimate image_proc_nodelet_manager">
        <param name="decimation_x" type="int" value="$(arg decim_fac)" />
        <param name="decimation_y" type="int" value="$(arg decim_fac)" />
        <param name="interpolation" type="int" value="1" />

        <!-- remap input topics -->
        <remap from="camera/image_raw" to="basler_camera/image_raw"/>
        <remap from="camera/image_info" to="basler_camera/camera_info"/>
        <!-- remap output topics -->
        <remap from="camera_out/image_raw" to="basler_camera_ds/image_raw"/>
        <remap from="camera_out/image_info" to="basler_camera_ds/camera_info"/>
    </node>

    <!-- Debayer and rectify -->
    <node pkg="nodelet" type="nodelet" name="image_proc_debayer" args="load image_proc/debayer image_proc_nodelet_manager">
        <!-- remap input topics -->
        <remap from="image_raw" to="basler_camera_ds/image_raw"/>

        <!-- remap output topics -->
        <remap from="image_mono" to="basler_camera_ds/image_mono"/>
        <remap from="image_color" to="basler_camera_ds/image_color"/>
    </node>
    <node pkg="nodelet" type="nodelet" name="image_proc_rectify" args="load image_proc/rectify image_proc_nodelet_manager" >
        <!-- remap input topics -->
        <remap from="image_mono" to="basler_camera_ds/image_mono"/>
        <remap from="camera_info" to="basler_camera_ds/camera_info"/>

        <!-- remap output topics -->
        <remap from="image_rect" to="basler_camera_ds/image_rect"/>
    </node>


</launch>
